### DISCUSSION TITLE
Allow renaming of table columns in Entries panel

### DISCUSSION THREAD

**Discussion by turnstylerj** [craftcms/cms#10046] [Ideas] [answered]
Being able to rename fields in the field layout designer allows for more generic fields to be created, like "Plain Text" or "Lightswitch" and used in various contexts with unique names, which helps a lot with consistency.

With this setup, however, if you'd like to set one of those fields as a table column in the Entries panel, you'll only get its generic name with no option to customize the label, so a one-off field with the specific title you want would need to be created.

It would be nice if each available table column had an edit icon along with its move handle where you could customize the name, as you can in the field layout designer.

Edit: @thupsi suggested a much better solution below. The label for the table column should just automatically inherit any custom label applied for that field in its entry type, no additional edit option needed.

**Comment by thupsi** [comment]
That's a good point! 

Maybe it would be simpler if the available table column names where mapped to the field names in the field layout designer for any given section?

So you would need to name the field only in one place per entry type.

The only complication I can think with this approach is that multiple columns per field would potentially need to exist if a field had different custom names for different entry types in the same section. That would be a technical complication but from a user point of view I don't think it would be a problem.

**Reply by turnstylerj** [reply]
Ah yes, that's a much better solution than allowing an additional manual edit. The label in the table column should just match any custom label for that field in the given entry type.

**Comment by brandonkelly** [comment, marked as answer]
This isn’t possible (without a setting a la “Header Column Heading” anyway) because any given element index source could be including multiple entries with the same field, but across multiple field layouts, each with unique custom labels for the field.

**Reply by thupsi** [reply]
The element index table could be tweaked to include as columns all custom labels from all entry types in the current source though? I'm certainly not in a position to say that would be simple to implement, just saying that it would be a nice little feature.
